app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt

```diff
@@ -193,7 +193,7 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
                 // The Browser fragment handles back presses on its own because it might just go back
                 // in the browsing history.
                 return
-            }
+            } else finish()
```

Comment:
Looking at the below code, if we remove the
```
if (!browserOverlay.isVisible) {
    // If the browser overlay is not visible then we show it now immediately. Depending on the loading
    // state of the page we m
    setOverlayVisible(true)
```
is this still necessary, or does the Activity handle the back behavior correctly?